lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -1,93 +1,99 @@
 #include "Ohmbrewer_Thermostat.h"
 #include "Ohmbrewer_Screen.h"
 #include "Ohmbrewer_Publisher.h"
+#include "pid.h"
 
+//Define Variables we'll be connecting to with PID
+double setPoint;
+double input;
+double output;
 
+//Define the aggressive and conservative Tuning Parameters for PID
+double aggKp=4;
+double aggKi=0.2;
+double aggKd=1;
+double consKp=1;
+double consKi=0.05;
+double consKd=0.25;
 
-/**
- * The desired target temperature. Defaults to Celsius
- * @returns The target temperature in Celsius, as a Temperature object pointer
- */
-Ohmbrewer::Temperature* Ohmbrewer::Thermostat::getTargetTemp() const {
-    return _targetTemp;
-}
-
-/**
- * Sets the target temperature
- * @param targetTemp The new target temperature in Celsius
- * @returns The time taken to run the method
- */
-const int Ohmbrewer::Thermostat::setTargetTemp(const double targetTemp) {
-    unsigned long start = millis();
-
-    _targetTemp->set(targetTemp);
+//PID windowing variables
+int windowSize = 5000;              //changeable value
+unsigned long windowStartTime;
 
-    return start - millis();
-}
+//PID set up
+PID* thermPID;
 
-/**
- * The Thermostat's heating element
- * @returns The heating element
- */
-Ohmbrewer::HeatingElement* Ohmbrewer::Thermostat::getElement() const {
-    return _heatingElm;
-}
-
-/**
- * The Thermostat's temperature sensor
- * @returns The temperature sensor
- */
-Ohmbrewer::TemperatureSensor* Ohmbrewer::Thermostat::getSensor() const {
-    return _tempSensor;
-}
 
 /**
- * Constructor
+ * Constructor - creating a Thermostat will use up 4 equipment IDs
  * @param id The Sprout ID to use for this piece of Thermostat
- * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
+ * @param thermPins list with formatting of: [ temp busPin ;  heating controlPin ; heating powerPin ]
+ * TODO add UID?
  */
-Ohmbrewer::Thermostat::Thermostat(int id, int (&thermPins)[3]) : Ohmbrewer::Equipment(id) {
-    int n = sizeof(thermPins) / sizeof(int);
-    if ( n > 2){
-        _heatingElm = new HeatingElement(id+2, thermPins[1], thermPins[2]);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else if (n == 2 ){                     // no switch, only power% (SSR in parallel or always on other leg)
-        _heatingElm = new HeatingElement(id+2, thermPins[1], -1);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else{
-        //publish error
-
+Ohmbrewer::Thermostat::Thermostat(int id, std::list<int>* thermPins) : Ohmbrewer::Equipment(id) {
+    //PID set up
+    thermPID = (new PID(&input, &output, &setPoint, consKp, consKi, consKd, PID::DIRECT));
```

Comment:
Will we always use PID::DIRECT, or is that a mode or some such that should be included in the member variables for Thermostat?